src/Microsoft.ML.Transforms/CategoryImputer.cs

```diff
@@ -0,0 +1,1829 @@
+                            if (!success)
+                                throw new Exception(GetErrorDetailsAndFreeNativeMemory(errorHandle));
+                        }
+                    }
+
+                    success = CreateTransformerFromEstimatorHelper(estimatorHandler, out IntPtr transformer, out errorHandle);
+                    if (!success)
+                        throw new Exception(GetErrorDetailsAndFreeNativeMemory(errorHandle));
+
+                    return new TransformerEstimatorSafeHandler(transformer, DestroyTransformerHelper);
+                }
+            }
+        }
+
+        #endregion
+
+        #region Int8Column
+
+        internal sealed class Int8TypedColumn : TypedColumn<sbyte>
+        {
+            private TransformerEstimatorSafeHandler _transformerHandler;
+            internal Int8TypedColumn(string name, string source) :
+                base(name, source, typeof(sbyte).ToString())
+            {
+            }
+
+            [DllImport("Featurizers", EntryPoint = "CatImputerFeaturizer_int8_t_CreateEstimator", CallingConvention = CallingConvention.Cdecl), SuppressUnmanagedCodeSecurity]
+            private static extern bool CreateEstimatorCpp(out IntPtr estimator, out IntPtr errorHandle);
+
+            [DllImport("Featurizers", EntryPoint = "CatImputerFeaturizer_int8_t_DestroyEstimator", CallingConvention = CallingConvention.Cdecl), SuppressUnmanagedCodeSecurity]
+            private static extern bool DestroyEstimatorCpp(IntPtr estimator, out IntPtr errorHandle); // Should ONLY be called by safe handle
+
+            [DllImport("Featurizers", EntryPoint = "CatImputerFeaturizer_int8_t_CreateTransformerFromEstimator", CallingConvention = CallingConvention.Cdecl), SuppressUnmanagedCodeSecurity]
+            private static extern bool CreateTransformerFromEstimatorCpp(TransformerEstimatorSafeHandler estimator, out IntPtr transformer, out IntPtr errorHandle);
+            [DllImport("Featurizers", EntryPoint = "CatImputerFeaturizer_int8_t_DestroyTransformer", CallingConvention = CallingConvention.Cdecl), SuppressUnmanagedCodeSecurity]
+            private static extern bool DestroyTransformerCpp(IntPtr transformer, out IntPtr errorHandle);
+            internal override void CreateTransformerFromEstimator(IDataView input)
+            {
+                _transformerHandler = CreateTransformerFromEstimatorBase(input);
+            }
+
+            [DllImport("Featurizers", EntryPoint = "CatImputerFeaturizer_int8_t_CreateTransformerFromSavedData", CallingConvention = CallingConvention.Cdecl), SuppressUnmanagedCodeSecurity]
+            private static extern bool CreateTransformerFromSavedDataCpp(IntPtr data, IntPtr bufferSize, out IntPtr transformer, out IntPtr errorHandle);
+            internal override void CreateTransformerFromSavedData(byte[] data)
+            {
+                bool result;
+                GCHandle handle = GCHandle.Alloc(data, GCHandleType.Pinned);
```

Comment:
You can just use a `fixed` statement in C#. This is slightly faster than allocing and freeing a GCHandle. See the rest of our code (like in CpuMath) where we get a pointer for an array.

Here's an example, from one of our tests:

https://github.com/dotnet/machinelearning/blob/b861b5d64841cbe0f2c866ee7586872aac450a51/test/Microsoft.ML.CpuMath.PerformanceTests/NativePerformanceTests.cs#L27

But you get the idea.

Reply:
Changed all occurrences to `fixed`.